WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewModel.swift

```diff
@@ -273,46 +269,69 @@ private extension DashboardViewModel {
+
+        // Onboarding card.
+        // When not available, Onboarding card needs to be hidden from Dashboard and Customize
+        cards.append(
+            DashboardCard(type: .onboarding,
+                          isAvailable: canShowOnboarding,
+                          enabled: canShowOnboarding,
+                          status: canShowOnboarding ? .show : .hide)
+        )
+
+        // Performance and Top Performance cards (also known as Analytics cards).
+        // When not available, Analytics cards need to be hidden from Dashboard, but appear on Customize as "Unavailable"
+        cards.append(
+            DashboardCard(type: .performance,
+                          isAvailable: canShowAnalytics,
+                          enabled: canShowAnalytics,
+                          status: canShowAnalytics ? .show : .unavailable)
+        )
+        cards.append(
+            DashboardCard(type: .topPerformers,
+                          isAvailable: canShowAnalytics,
+                          enabled: canShowAnalytics,
+                          status: canShowAnalytics ? .show : .unavailable)
+        )
+
+        // Blaze card.
+        // When not available, Blaze card needs to be hidden from Dashboard and Customize
+        cards.append(
+            DashboardCard(type: .blaze,
+                          isAvailable: canShowBlaze,
+                          enabled: canShowBlaze,
+                          status: canShowBlaze ? .show : .hide)
+        )
+
+        return cards
+    }
+
     @MainActor
     func updateDashboardCards(canShowOnboarding: Bool,
                               canShowBlaze: Bool,
                               canShowAnalytics: Bool) async {
-        dashboardCards = await {
-            if var stored = await loadDashboardCards() {
-                let analyticCardTypes: [DashboardCard.CardType] = [.performance, .topPerformers]
-                stored = canShowAnalytics ? stored : stored.filter { !analyticCardTypes.contains($0.type) }
-                return stored
+
+        // First, generate latest cards state based on current canShow states
+        let initialCards = generateDefaultCards(canShowOnboarding: canShowOnboarding,
+                                                canShowBlaze: canShowBlaze,
+                                                canShowAnalytics: canShowAnalytics)
+
+        // Next, get saved cards and preserve existing enabled state for all available cards.
+        // This is needed because even if a user already disabled an available card and saved it, in `initialCards`
+        // the same card might be set to be enabled. To respect user's setting, we need to check the saved state and re-apply it.
+        let savedCards = await loadDashboardCards() ?? []
+        dashboardCards = initialCards.map { initialCard in
+            if let savedCard = savedCards.first(where: { $0.type == initialCard.type }),
+               savedCard.isAvailable && initialCard.isAvailable {
```

Comment:
Can you clarify the reasoning behind this check?

Reply:
I put some explanation for this here: https://github.com/woocommerce/woocommerce-ios/pull/12626/files#r1587606533